examples/commandShowcase.lean

```diff
@@ -0,0 +1,78 @@
+import Qq
+
+open Qq
+
+-- using the goal type (interpret 2, 3 as Int)
+/--
+info: x
+---
+info: Prop
+-/
+#guard_msgs in
+def f (x : Prop) [Decidable x] : Int :=
+  by_elabq
+    Lean.logInfo x
+    Lean.logInfo x.ty
+    return q(if $x then 2 else 3)
```

Comment:
Does this work?
```suggestion
macro "trace_state" : doElem => `(doElem| (by trace_state; exact pure ()))

-- using the goal type (interpret 2, 3 as Int)
#guard_msgs in
def f (x : Prop) [Decidable x] : Int :=
  by_elabq
    trace_state
    return q(if $x then 2 else 3)
```

Reply:
Yes, that works, not sure if `trace_state` is the best name.

Reply:
It's local to the test file so I wouldn't worry too much

Reply:
What matters is that this should let us see what lean elaborated the goal type to!

Reply:
I would also change it to ``(doElem| pure (by trace_state; exact ()))` to have shorter goal.

Reply:
I think it's better showing the current monad to make the comments match what the editor shows.